backend/cmd/backend/main.go

```diff
@@ -1,21 +1,63 @@
 package main
 
 import (
+	"encoding/json"
+	"flag"
+	"fmt"
 	"log"
+	"net"
+	"strconv"
 
 	nosql "../../pkg/nosql"
 )
 
+//Packet will hold JSON information for later decoding
+type Packet struct {
+	Action   string `json:"action"`
+	CompName string `json:"compName"`
+}
+
+func handleRequest(c net.Conn) {
+	var header Packet
+	// we create a decoder that reads directly from the socket
+	decode := json.NewDecoder(c)
+
+	decodeErr := decode.Decode(&header)
+	if decodeErr != nil {
+		log.Println(decodeErr)
+	}
+
+	fmt.Printf("Action: %s, CompName: %s\n", header.Action, header.CompName)
+
+	c.Close()
```

Comment:
Defer this at the start of the function instead and change the net.Conn parameter to take a pointer

```go
func handleRequest(c *net.Conn) {
    defer c.Close()
    ...
}
```